README.md

```diff
@@ -215,3 +192,379 @@ if available(.settingsFeature) {
+
+    func buyCrypto(_ crypto: Crypto) {...}
+}
+```
+
+🔴
+```
+final class SomeClass {
+
+    // MARK: - Init
+    public init() {...}
+
+    // MARK: - BuyService
+    func buyCrypto(_ crypto: Crypto) {...}
+}
+```
+
+
+If a large number of private functions are typed, then you should not dump them all in one pile, you need to break them into logically connected blocks.
+
+🟢 Division within the class:
+```
+final class SomeClass {
+
+    // MARK: - Listeners
+
+    private func addListeners() {...}
+    private func removeListeners() {...}
+    private func removeListener(_ listener: Listener) {...}
+
+    // MARK: - Actions
+
+    @objc private func didTapClose() {...}
+}
+```
+
+
+🟢 Takeaway in extensions:
+```
+final class SomeClass {
+    ...
+}
+
+// MARK: - Listeners
+
+extension SomeClass {
+    private func addListeners() {...}
+    private func removeListeners() {...}
+    private func removeListener(_ listener: Listener) {...}
+}
+
+// MARK: - Actions
+
+extension SomeClass {
+   @objc private func didTapClose() {...}
+}
+```
+
+🔴
+```
+final class SomeClass {
+
+    // MARK: - Private
+
+    func addListeners() {...}
+    @objc func didTapClose() {...}
+    func removeListeners() {...}
+    func removeListener(_ listener: Listener) {...}
+}
+```
+
+
+There should be no line break before the closing brackets:
+
+🟢
+```
+struct SomeStruct {
+    func listen() {
+        ...
+    }
+}
+```
+
+🔴
+```
+struct SomeStruct {
+    func listen() {
+        ...
+    }
+
+}
+```
+
+
+### Switch statement
+For enum we don't use default in switch. When changing the enum during assembly, all the places where it is used will be immediately visible:
+
+🟢
+```
+enum SomeEnum {
+    case one
+    case two
+    case three
+    case four
+}
+
+switch enum {
+    case .one: // do something
+    case .two: // do something
+    case .three, .four: break
+}
+```
+
+🔴
+```
+enum SomeEnum {
+    case one
+    case two
+    case three
+    case four
+}
+
+switch enum {
+    case .one: // do something
+    case .two: // do something
+    default: break
+}
+```
+
+
+### Redundant code
+In the .map functions .filter .reduce etc. omit the parentheses:
+
+🟢
+```
+dict.map { $0 }
+dict.filter { $0 % 2 == 0 }
+```
+
+🔴
+```
+dict.map({ $0 })
+dict.filter({ $0 % 2 == 0 })
+```
+
+
+For the returned parameters in closure, we omit the parentheses:
+
+🟢
+```
+let handler: SomeHandler = { [weak self] action, indexPath in
+    self?.didTrigger(action, onItemAt: indexPath)
+}
+```
+
+🔴
+```
+let handler: SomeHandler = { [weak self] (action, indexPath) in
+    self?.didTrigger(action, onItemAt: indexPath)
+}
+```
+
+
+### TODO comments
+In TODO, we specify the version in which the fix is planned, your nickname and a link to the task in JIRA.
+
+```
+// TODO: 2.7 vasya.pupkin later take out the logic in BuyService https://jira..../task
+```
+
+
+### Constants
+### Local constants
+All constants should be at the very top of the file, right after the imports.
```

Comment:
I think the opposite. Constants should be at the BOTTOM of the file. Think about the SomeView.swift. When opening the SomeView.swift, we expect to see the View first, not the constant.

Reply:
But it's comfortable to manage them from the top. Why do you think about this @lisemyon @sidorov-panda @TrGiLong

Reply:
Agree, at the top is better